bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/PiHoleHandler.java

```diff
@@ -0,0 +1,292 @@
+        updateDecimalState(ADS_BLOCKED_TODAY_CHANNEL, localDnsStatistics.getAdsBlockedToday());
+        updateDecimalState(UNIQUE_DOMAINS_CHANNEL, localDnsStatistics.getUniqueDomains());
+        updateDecimalState(QUERIES_FORWARDED_CHANNEL, localDnsStatistics.getQueriesForwarded());
+        updateDecimalState(QUERIES_CACHED_CHANNEL, localDnsStatistics.getQueriesCached());
+        updateDecimalState(CLIENTS_EVER_SEEN_CHANNEL, localDnsStatistics.getClientsEverSeen());
+        updateDecimalState(UNIQUE_CLIENTS_CHANNEL, localDnsStatistics.getUniqueClients());
+        updateDecimalState(DNS_QUERIES_ALL_TYPES_CHANNEL, localDnsStatistics.getDnsQueriesAllTypes());
+        updateDecimalState(REPLY_UNKNOWN_CHANNEL, localDnsStatistics.getReplyUnknown());
+        updateDecimalState(REPLY_NODATA_CHANNEL, localDnsStatistics.getReplyNoData());
+        updateDecimalState(REPLY_NXDOMAIN_CHANNEL, localDnsStatistics.getReplyNXDomain());
+        updateDecimalState(REPLY_CNAME_CHANNEL, localDnsStatistics.getReplyCName());
+        updateDecimalState(REPLY_IP_CHANNEL, localDnsStatistics.getReplyIP());
+        updateDecimalState(REPLY_DOMAIN_CHANNEL, localDnsStatistics.getReplyDomain());
+        updateDecimalState(REPLY_RRNAME_CHANNEL, localDnsStatistics.getReplyRRName());
+        updateDecimalState(REPLY_SERVFAIL_CHANNEL, localDnsStatistics.getReplyServFail());
+        updateDecimalState(REPLY_REFUSED_CHANNEL, localDnsStatistics.getReplyRefused());
+        updateDecimalState(REPLY_NOTIMP_CHANNEL, localDnsStatistics.getReplyNotImp());
+        updateDecimalState(REPLY_OTHER_CHANNEL, localDnsStatistics.getReplyOther());
+        updateDecimalState(REPLY_DNSSEC_CHANNEL, localDnsStatistics.getReplyDNSSEC());
+        updateDecimalState(REPLY_NONE_CHANNEL, localDnsStatistics.getReplyNone());
+        updateDecimalState(REPLY_BLOB_CHANNEL, localDnsStatistics.getReplyBlob());
+        updateDecimalState(DNS_QUERIES_ALL_REPLIES_CHANNEL, localDnsStatistics.getDnsQueriesAllTypes());
+        updateDecimalState(PRIVACY_LEVEL_CHANNEL, localDnsStatistics.getPrivacyLevel());
+
+        var adsPercentageToday = localDnsStatistics.getAdsPercentageToday();
+        if (adsPercentageToday != null) {
+            var state = new QuantityType<>(new BigDecimal(adsPercentageToday.toString()), PERCENT);
+            updateState(ADS_PERCENTAGE_TODAY_CHANNEL, state);
+        }
+        updateState(ENABLED_CHANNEL, OnOffType.from(localDnsStatistics.getEnabled()));
+        if (localDnsStatistics.getEnabled()) {
+            updateState(DISABLE_ENABLE_CHANNEL, new StringType(ENABLE.toString()));
+        }
+    }
+
+    private void updateDecimalState(String channelID, @Nullable Integer value) {
+        if (value == null) {
+            return;
+        }
+        updateState(channelID, new DecimalType(value));
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(PiHoleActions.class);
```

Comment:
Can be replaced by `Set.of(PiHoleActions.class)`

Reply:
Documentation could be updated, usage has evolved [since then ](https://github.com/search?q=repo%3Aopenhab%2Fopenhab-addons%20Collection%3CClass%3C%3F%20extends%20ThingHandlerService%3E%3E%20getServices()&type=code)